pkg/crc/network/nameservers.go

```diff
@@ -32,15 +34,51 @@ func GetResolvValuesFromInstance(sshRunner *ssh.Runner) (*ResolvFileValues, erro
 }
 
 func CreateResolvFileOnInstance(sshRunner *ssh.Runner, resolvFileValues ResolvFileValues) error {
-	resolvFile, _ := CreateResolvFile(resolvFileValues)
+	sd := systemd.NewInstanceSystemdCommander(sshRunner)
+	// Check if ovs-configuration.service exist and if not then it is old bundle and use the same way to
+	// update resolve.conf file
+	if state, err := sd.Status("ovs-configuration.service"); err != nil || state == states.NotFound {
+		if err := replaceResolvConfFile(sshRunner, resolvFileValues); err != nil {
+			return fmt.Errorf("error updating resolve.conf file: %s", err)
+		}
+		return nil
+	}
+
+	if err := sd.Start("ovs-configuration.service"); err != nil {
+		return err
+	}
+
+	if err := updateNetworkManagerConfig(sshRunner, resolvFileValues); err != nil {
+		return fmt.Errorf("failed to update NetworkManager configuration: %s", err)
+	}
+
+	return sd.Restart("NetworkManager.service")
+}
 
-	err := sshRunner.CopyDataPrivileged([]byte(resolvFile), "/etc/resolv.conf", 0644)
+func replaceResolvConfFile(sshRunner *ssh.Runner, resolvFileValues ResolvFileValues) error {
+	resolvFile, err := CreateResolvFile(resolvFileValues)
+	if err != nil {
+		return fmt.Errorf("error to create resolv conf file: %v", err)
+	}
+	err = sshRunner.CopyDataPrivileged([]byte(resolvFile), "/etc/resolv.conf", 0644)
 	if err != nil {
 		return fmt.Errorf("Error creating /etc/resolv on instance: %s", err.Error())
 	}
 	return nil
 }
 
+func updateNetworkManagerConfig(sshRunner *ssh.Runner, resolvFileValues ResolvFileValues) error {
+	nameservers := strings.Join(resolvFileValues.GetNameServer(), ",")
+	searchDomains := strings.Join(resolvFileValues.GetSearchDomains(), ",")
+	// When ovs-configuration service is running we are sure that name of the connection is ovs-if-br-ex
```

Comment:
I'd replace "we are sure" with "should be" or "is expected to be" :)

Reply:
Done